codegen/test/common.hpp

```diff
@@ -8,18 +8,73 @@
 #include <rtc/compile_kernel.hpp>
 #include <rtc/hip.hpp>
 #include <fstream>
+#include <unordered_set>
+#include "ck/host/headers.hpp"
+#include "rtc/hiprtc_enable_env.hpp"
+#include "ck/host/stringutils.hpp"
 
-std::vector<rtc::src_file> get_headers_for_test()
+// NOLINTNEXTLINE
+const char* const disable_warning_pragma = R"__migraphx__(
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Weverything"
+${content}
+#pragma clang diagnostic pop
+)__migraphx__";
+
+template <class P>
+inline std::string ck_disable_warnings(P p)
+{
+    return ck::host::InterpolateString(disable_warning_pragma,
+                                       {{"content", std::string{p.data(), p.size()}}});
+}
```

Comment:
This is not needed as we shouldn't be enabling `-Werror` on the tests(ck files are already checked for warnings in the main build).